File: src/python/test_boltz_jones.py
from src.python.boltz_jones import BeatDetector
import time

detector = BeatDetector()

print("Simulation de 4 battements espacés de 500ms (BPM attendu ≈ 120)")
for _ in range(4):
    time.sleep(0.5)  # 500 ms
    result = detector.process_beat()
    print(f"BPM: {result['bpm']}, Phase offset: {result['phase_offset_ms']} ms, Timestamp: {result['timestamp']}, Confidence: {result['confidence']}")

File: src/rust/test_boltz_jones.rs
mod boltz_jones;
use std::thread;
use std::time::Duration;

fn main() {
    let mut detector = boltz_jones::BoltzJonesDetector::new(0);

    println!("Simulation de 4 battements espacés de 500ms (BPM attendu ≈ 120)");
    for _ in 0..4 {
        thread::sleep(Duration::from_millis(500));
        let (bpm, phase) = detector.process_beat();
        println!("BPM: {:.2}, Phase offset: {} ms", bpm, phase);
    }
}
